### PR TITLE
feat: update mwe.md to lean4

### PR DESCRIPTION
**Pull request by joneugster**
Update mwe.md to match the lean4 infrastructure.

### CODE REVIEW

File: templates/mwe.md [hunk 1]
```diff
@@ -1,17 +1,5 @@
...
-</p>
-<p>
-Pull requests updating this page for Lean 4 are very welcome.
-There is a link at the bottom of this page.
-</p>
-<p>
-Please visit <a href="https://leanprover.zulipchat.com">the leanprover zulip</a>
-and ask for whatever help you need during this transitional period!
-</p>
-<p>
```
Comment (PatrickMassot) [inline templates/mwe.md:14]: I think you need to remove the entire div starting on line 1.

File: templates/mwe.md [hunk 2]
```diff
@@ -21,26 +9,30 @@ If you need to link to Lean 3 specific resources please link there.
 
 ## tl;dr
 
-When posting code on Zulip, please include all `import`s and `open`s and `universe`s and `variable`s, so others can simply just cut and paste what you post, and see the same issue that you are seeing. The best way to ensure you have done this is to make a completely new Lean file containing only what you are proposing to post, and checking that it compiles.
+When posting code on Zulip, please include all `import`s, `open`s, `universe`s, and `variable`s, so others can simply just copy-paste what you post, and see the same issue that you are seeing.
+
+The best way to ensure you have done this is to copy-paste the code snipped you are about to post into the [lean web editor](https://lean.math.hhu.de) (or an empty Lean file) and check it compiles.
```
Comment (PatrickMassot) [inline templates/mwe.md:14]: How often is the lean web editor mathlib updated? Using it seems more complicated than using an empty Lean file, so I would use it as our first recommendation.

File: templates/mwe.md [hunk 3]
```diff
@@ -21,26 +9,30 @@ If you need to link to Lean 3 specific resources please link there.
...
 ```lean
-#check (univ : set X)
+#check (univ : Set X)
 ```
 
 ### Good example:
 
 ```lean
-import data.set.basic
+import Mathlib
```
Comment (PatrickMassot) [inline templates/mwe.md:27]: Why importing everything?

File: templates/mwe.md [hunk 4]
```diff
@@ -58,21 +50,23 @@ h2 : a.fst < b.snd
...
+  -/
+  done
 ```
 
-Tip: If you are using [mathlib](https://github.com/leanprover-community/mathlib) and have `import tactic` in your file, there's a tactic called [`extract_goal`](https://leanprover-community.github.io/mathlib_docs/tactics.html#extract_goal) that can help you format the current goal as a stand-alone example. You can remove extraneous variables and hypotheses from the output of `extract_goal` to further minimize your example.
+<!---
+TODO: update link to `extract_goal` once the lean4-equivalent of `tactics.html` exists.
+-->
+
+Tip: If you are using [mathlib](https://github.com/leanprover-community/mathlib4) (e.g. with `import Mathlib`), there's a tactic called [`extract_goal`](https://leanprover-community.github.io/mathlib_docs/tactics.html#extract_goal) that can help you format the current goal as a stand-alone example. You can remove extraneous variables and hypotheses from the output of `extract_goal` to further minimize your example.
```
Comment (PatrickMassot) [inline templates/mwe.md:69]: Please link to https://leanprover-community.github.io/mathlib4_docs/Mathlib/Tactic/ExtractGoal.html

File: templates/mwe.md [hunk 5]
```diff
@@ -91,24 +85,25 @@ It is fine for your example to throw compiler errors or warnings. In particular,
 
 ## How do I know if my code is a MWE?
 
-You should *test* this by making a new Lean file, pasting your code snippet into it, and seeing if you get the expected behavior. This is exactly what people who try to help you will do!
+You should *test* this by copy-pasting your code snippet into the [lean web editor](https://lean.math.hhu.de), or into a new Lean file, and seeing if you get the expected behavior. This is exactly what people who try to help you will do!
```
Comment (PatrickMassot) [inline templates/mwe.md:88]: Same remark about the web editor.

File: templates/mwe.md [hunk 6]
```diff
@@ -91,24 +85,25 @@ It is fine for your example to throw compiler errors or warnings. In particular,
...
 
-## What if I'm asking about the [Natural Number Game](https://www.ma.imperial.ac.uk/~buzzard/xena/natural_number_game/)?
+## What if I'm asking about the [Natural Number Game](https://adam.math.hhu.de/)?
```
Comment (PatrickMassot) [inline templates/mwe.md:90]: Why not https://adam.math.hhu.de/#/g/hhu-adam/NNG4?

**Review by PatrickMassot — COMMENTED** [review]
Thanks for working on this!

File: templates/mwe.md (hunk 3 above)
Comment (joneugster) [reply templates/mwe.md:27]: Simplicity. I decided to add the instruction below that `import Mathlib` is perfectly fine in a mwe. My reasons were the following:

- wasn't possible in lean3 (unless you created the all.lean manually first), which might be why it wasn't used.
- is much simpler for beginners than searching the correct import
- ensures that you are not just simply missing imports (e.g. if some instances are not in `XY/Basic.lean`)
- Recent conversations on Zulip do use `import Mathlib` on a frequent basis in the MWEs.

File: templates/mwe.md (hunk 1 above)
Comment (joneugster) [reply templates/mwe.md:14]: I thought it might be useful to still have the link to the archived lean3 page for some time, but I can remove it.

File: templates/mwe.md (hunk 6 above)
Comment (joneugster) [reply templates/mwe.md]: I don't have a strong opinion about linking the landing page, Ill change it.

File: templates/mwe.md (hunk 5 above)
Comment (joneugster) [reply templates/mwe.md]: done

File: templates/mwe.md (hunk 4 above)
Comment (joneugster) [reply templates/mwe.md]: done

File: templates/mwe.md (hunk 2 above)
Comment (joneugster) [reply templates/mwe.md]: It's currently updated every 6h, which we might change according to the demand, but it'll stay updated minimum once a day.

I thought that was less complicated, but I'm happy to have it changed around in the text.

File: templates/mwe.md (hunk 1 above)
Comment (PatrickMassot) [reply templates/mwe.md:14]: No, we want people to forget about Lean 3. Please remove that div.

**Comment by PatrickMassot** [comment]
Thanks!
